### PR TITLE
Added vexo analytics as an option for the analytics providers.

### PR DESCRIPTION
**Pull request by falcoagustin**
## Feature Request

We'd like to add [vexo](https://www.vexo.co/) as an option when creating an app in any of the possible configurations.

To do so:
* I've modified the corresponding files to support any analytics tool
* Added vexo as an option in the CLI
* Modified base index.ts file for Expo, so that we can use `ejs` in it to generate corresponding files
* Included the code in `package.json` and added the code for React Navigation

### Notes

I took the liberty of indenting correctly part of the code generated since at first glance, I thought it could be improved and a bit messy. In addition, we should solve the issue when we are generating ejs files, `if` statements that are not met will leave a blank line.

Other than that, thank you very much!
Eager to hear your thoughts

### CODE REVIEW

**Comment by danstepanov** [comment]
@falcoagustin Is there any configuration you'd like to add to projects beyond getting set up with an API key and a vexo account? As an FYI, we're about to have a documentation section on the website or we can simply link to your documentation.

**Comment by danstepanov** [comment]
Additionally, this PR will need to be rebased with `main` given the new monorepo structure, @ernestoresende would you mind helping out with this?

**Comment by ernestoresende** [comment]
For sure, but I'll have to be added as a direct collaborator to the repo and this PR should have [Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) turned on in order for me to contribute directly with commits from my end.

Alternatively, I can guide you and/or @falcoagustin on how to resolve the merge conflicts on your end.

**Comment by danstepanov** [comment]
> For sure, but I'll have to be added as a direct collaborator to the repo and this PR should have [Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) turned on in order for me to contribute directly with commits from my end.
> 
> Alternatively, I can guide you and/or @falcoagustin on how to resolve the merge conflicts on your end.

Invited

**Comment by falcoagustin** [comment]
@ernestoresende @danstepanov No further config I'd like on the tool. 
Initially it's a good starting point, we can then build in top of this if needed, as I'd like to keep things as simple as possible, not only for Vexo, but for other providers that are integrating with us. Everything should be as smooth as possible.

**Comment by ernestoresende** [comment]
All good for me, I'll leave final review and merge to @danstepanov.

Thanks for your contribution and your patience with our monorepo shenanigans @falcoagustin 🚀

**Comment by falcoagustin** [comment]
Sure thing!
Thank you very much let me know if I can do anything else to help out

File: cli/src/templates/base/package.json.ejs
Comment (danstepanov) [inline cli/src/templates/base/package.json.ejs:6]: @falcoagustin can you include these changes in your PR? @ernestoresende when i was testing, I think this change was necessary while using bun

File: cli/src/templates/base/package.json.ejs
Comment (danstepanov) [inline cli/src/templates/base/package.json.ejs:32]: @falcoagustin this needs to remain `^4.6.3`

File: cli/src/templates/base/package.json.ejs
Comment (danstepanov) [inline cli/src/templates/base/package.json.ejs:58]: @falcoagustin this needs to remain

File: cli/src/templates/base/package.json.ejs
Comment (danstepanov) [inline cli/src/templates/base/package.json.ejs:64]: @falcoagustin this also needs to remain

File: cli/src/templates/base/package.json.ejs
Comment (danstepanov) [inline cli/src/templates/base/package.json.ejs:97]: @falcoagustin this also needs to remain

File: cli/src/templates/base/package.json.ejs
Comment (danstepanov) [inline cli/src/templates/base/package.json.ejs:102]: @falcoagustin same here

File: cli/src/commands/create-expo-stack.ts
Comment (danstepanov) [inline cli/src/commands/create-expo-stack.ts:130]: @falcoagustin can you add this to the --help option? should be here https://github.com/danstepanov/create-expo-stack/blob/main/cli/src/utilities/showHelp.ts

**Review by danstepanov — CHANGES_REQUESTED** [review]
almost ready!

**Comment by ernestoresende** [comment]
@danstepanov 

Being fair to @falcoagustin, some of the removed things that shouldn't have been removed might be a result of my rebase and subpar Git skills. That's on me.

From what I can see of his original commits, the only thing changed was indentation, ex.: 

https://github.com/danstepanov/create-expo-stack/blob/61eaf1240b1fc102286159e585e597d4edef2181/src/templates/base/package.json.ejs#L23-L28

Not exactly sure what I might have done during the merge that caused this to happen, but I'll try restoring it to the original state if @falcoagustin is busy to take a look at this today.

**Comment by falcoagustin** [comment]
That is correct @ernestoresende , none of my changes did change versions but only indentation since for the end project the indentation there was incorrect and untidy. That's also explained under notes section @danstepanov
Will fix soon! Thanks for reviewing

**Comment by falcoagustin** [comment]
If @ernestoresende you wanna take it today, feel free to!
I'll take a peek over the weekend if that's okay

**Comment by sammoore** [comment]
Think I have [a better / more up to date rebase](https://github.com/danstepanov/create-expo-stack/compare/main...sammoore:create-expo-stack:falcoagustin/add-vexo-analytics)!

**Comment by alejorod** [comment]
Update the PR to reflect the latest changes from main and added extra steps on success to guide the users with the vexo integration. Also added the `vexo-analytics` to `--help` and `.env` vexo props. 

Let us know if anything else is needed

**Comment by danstepanov** [comment]
@alejorod will review/test this in the morning, thanks!

File: cli/src/utilities/configureProjectFiles.ts
Comment (danstepanov) [inline cli/src/utilities/configureProjectFiles.ts:249]: i'm not sure this is would work. Doing this for firebase and supabase works because you can't pick both, only one so only one `.env` is added. However, with this line, you might end up with an overwritten `.env` file. Further, how would that file account for firebase or supabase?

I would suggest adding this check after the firebase and supabase additions and then first checking whether a .env has already been added. If so, do nothing and simply modify the existing ones via EJS file generation. If not, add a `.env` with the vars required for Vexo.

File: cli/src/utilities/printOutput.ts
Comment (danstepanov) [inline cli/src/utilities/printOutput.ts:137]: is this `info(``)` necessary if `printVexoSteps()` already has the padding?

File: cli/src/utilities/printOutput.ts
Comment (danstepanov) [inline cli/src/utilities/printOutput.ts:140]: is this `info(``)` necessary if `printVexoSteps()` already has the padding?

File: cli/src/utilities/runCLI.ts
Comment (danstepanov) [inline cli/src/utilities/runCLI.ts:69]: I think having Vexo as an option flag makes the most sense rather than a prompt question. I am concerned with the CLI having too many questions.

**Comment by danstepanov** [comment]
@alejorod I provided feedback, can you also remove the changes that are just deleting empty lines? Not sure why our linter isn't catching those.

File: cli/src/utilities/printOutput.ts
Comment (alejorod) [reply cli/src/utilities/printOutput.ts:137]: This one is not necessary, Im removing it.

File: cli/src/utilities/printOutput.ts
Comment (alejorod) [reply cli/src/utilities/printOutput.ts:140]: This one happens after the success message, outside of the Vexo steps. I think we should keeping it for consistency as the other paths (supabase, firebase) include that last padding.

File: cli/src/utilities/runCLI.ts
Comment (alejorod) [reply cli/src/utilities/runCLI.ts:69]: We can remove it if necessary. I think there is a benefit of having all third party libraries choices on both ways of initializing your app, the prompt flow works especially well for Vexo since it requires a navigation library to be present.

File: cli/src/utilities/configureProjectFiles.ts
Comment (alejorod) [reply cli/src/utilities/configureProjectFiles.ts]: I've added the firebase/supabase to the vexo `.env` through the `include` tag instead of hardcoding them.
Are you suggesting we use the `include` on the firebase/supabase files instead?

As an alternative we could have a `.env` file on `templates/base` which include other `.env.ejs` depending on the packages included, this would make future integrations which need to set env variables easier to setup

**Comment by alejorod** [comment]
hey @danstepanov thanks for the feedback. I update the PR with those changes and left some comments regarding the prompt and `.env` files

File: cli/src/utilities/runCLI.ts
Comment (danstepanov) [reply cli/src/utilities/runCLI.ts:69]: Your concern for the prerequisite of a navigation library is valid. I think that is addressable via an conditional check for whether a navigation package is present. 

My concern re adding Vexo to the core CLI prompts stems largely from CES' stated goal. CES is meant to be a rapid iteration tool rather than as a way to initialize a formal project (hence why we created https://shiprn.dev). Subsequently, the core CLI prompts should only include questions that are necessary for rapid iteration (styling and navigation). I'm even considering removing the auth chunk from CES. I'm happy to add Vexo to ShipRN which is a lot more opinionated.

If you believe there is a case to be made for how analytics are core to rapid iteration, I'm here for it.

File: cli/src/utilities/printOutput.ts
Comment (danstepanov) [reply cli/src/utilities/printOutput.ts]: I don't disagree but we should ensure that there is only 1 space, for visual consistency. Where you remove said space from is up to you.

File: cli/src/utilities/configureProjectFiles.ts
Comment (danstepanov) [reply cli/src/utilities/configureProjectFiles.ts]: Good points. I think having a single `.env.ejs` in `templates/base` is a solid idea but I think it should be excluded from the generated project if the user selects a configuration that does not require it.

**Review by danstepanov — APPROVED** [review]
LGTM

**Comment by danstepanov** [comment]
@falcoagustin once this is rebased with `main`, we can merge!

**Comment by alejorod** [comment]
@danstepanov rebase done! Let me know if anyhting else is missing

File: cli/src/types.ts
Comment (danstepanov) [inline cli/src/types.ts:29]: @sammoore As this adds a new option to the CLI, should we add something to that file you mentioned such that this doesn't break https://rn.new?

File: cli/src/types.ts
Comment (danstepanov) [inline cli/src/types.ts:65]: @dannyhw We'll want to add this into #325

**Comment by danstepanov** [comment]
@alejorod Thanks for updating the PR! Couple nits:
1) Please remove any changes which solely affect the format of the code
2) Awaiting confirmation from @sammoore and @dannyhw re any additional changes needed due to work over the last week

Otherwise, everything LGTM

File: cli/src/types.ts
Comment (dannyhw) [reply cli/src/types.ts:65]: ok no problem

**Review by danstepanov — CHANGES_REQUESTED** [review]
see above comments

**Comment by alejorod** [comment]
@danstepanov reverted changes which solely affect the format of the code

**Review by danstepanov — APPROVED** [review]
LGTM

Will need to (1) add one thing to a validation file and (2) add to our analytics but I'll do that. This should be available in either the next patch version of CES or the next minor version. Either way, it will likely be publiushed around Monday June 24th.

Once I merge this, you can test it via `npx create-expo-stack@next --vexo-analytics`

**Pull request merged**
